src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageExtensions.cs

```diff
@@ -52,6 +54,25 @@ public static IApplicationBuilder UseDeveloperExceptionPage(this IApplicationBui
             throw new ArgumentNullException(nameof(options));
         }
 
-        return app.UseMiddleware<DeveloperExceptionPageMiddleware>(Options.Create(options));
+        return SetMiddleware(app, options);
+    }
+
+    private static IApplicationBuilder SetMiddleware(
```

Comment:
Why do we need this change?

Reply:
Because we added a new constructor overload to the `Middleware`, the instance is create using a custom `ActivatorUtilities` that uses the args to detect the best `constructor`, if I did not include the service here it will pick-up the first `declared` constructor returned by a call to `GetConstructors`. Adding the service, when registered we have guaranteed that the  `ActivatorUtilities` selects a constructor that contains a parameter with `IProblemDetailsService`.

https://github.com/dotnet/aspnetcore/blob/d6d2b0cae0599236696d9b3b0a8c42012377cab2/src/Shared/ActivatorUtilities/ActivatorUtilities.cs#L24